File: CS/GrowthWare.Framework/Model/Profiles/MDBInformation.cs
using GrowthWare.Framework.Model.Profiles.Base;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GrowthWare.Framework.Model.Profiles
{
    /// <summary>
    /// Base properties an DB Information Profile
    /// </summary>
    /// <remarks>
    /// Corresponds to table ZGWSystem.Database_Information and 
    /// Store procedures: 
    /// ZGWSystem.Set_DataBase_Information, ZGWSystem.Get_Database_Information
    /// </remarks>
    [Serializable(), CLSCompliant(true)]
    public class MDBInformation : MProfile
    {
#region "Member Properties"
        private int mInformationSeqId = 1;
        private string mVersion = string.Empty;
        private int mEnableInheritance = 1;
#endregion

#region "Public Properties"
        public int Information_SEQ_ID
        {
            get { return mInformationSeqId; }
            set { mInformationSeqId = value; }
        }

        public string Version
        {
            get { return mVersion.Trim(); }
            set { mVersion = value.Trim(); }
        }

        public int EnableInheritance
        {
            get { return mEnableInheritance; }
            set { mEnableInheritance = value; }
        }
#endregion

#region "Private Methods"
        protected new void Initialize(ref DataRow dataRow)
        {
            base.NameColumnName = "VERSION";
            base.IdColumnName = "Information_SEQ_ID";
            base.Initialize(dataRow);
            mInformationSeqId = base.GetInt(dataRow, "Information_SEQ_ID");
            mVersion = base.GetString(dataRow, "VERSION");
            mEnableInheritance = base.GetInt(dataRow, "ENABLE_INHERITANCE");
        }
#endregion

#region "Public Methods"

        /// <summary>
        /// Initializes a new instance of the <see cref="MDBInformation"/> class.
        /// </summary>
        public MDBInformation()
        {
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="MDBInformation"/> class.
        /// </summary>
        /// <param name="dataRow">The dataRow.</param>
        public MDBInformation(DataRow dataRow)
        {
            this.Initialize(ref dataRow);
        }
#endregion
    }
}

File: VB/GrowthWare.Framework/Model/Profiles/MDBInformation.vb
Imports GrowthWare.Framework.Model.Profiles.Base

Namespace Model.Profiles
    ''' <summary>
    ''' Class MDBInformation.
    ''' </summary>
    <System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1709:IdentifiersShouldBeCasedCorrectly", MessageId:="MDB")>
    <Serializable(), CLSCompliant(True)> _
    Public Class MDBInformation
        Inherits MProfile

#Region "Constructors"
        ''' <summary>
        ''' Provides a new account profile with the default vaules
        ''' </summary>
        ''' <remarks></remarks>
        Public Sub New()

        End Sub

        ''' <summary>
        ''' Initializes a new instance of the <see cref="MDBInformation"/> class.
        ''' </summary>
        ''' <param name="dataRow">The data row.</param>
        Public Sub New(ByVal dataRow As DataRow)
            If Not dataRow Is Nothing Then
                Me.Initialize(dataRow)
            End If
        End Sub
#End Region

#Region "Member Properties"
        Private m_InformationSeqId As Integer = 1
        Private m_Version As String = String.Empty
        Private m_EnableInheritance As Integer = 1
#End Region


#Region "Public Properties"
        Public Property InformationSeqId() As Integer
            Get
                Return m_InformationSeqId
            End Get
            Set(ByVal value As Integer)
                m_InformationSeqId = value
            End Set
        End Property

        Public Property Version() As String
            Get
                Return m_Version.Trim
            End Get
            Set(ByVal value As String)
                If Not value Is Nothing Then m_Version = value.Trim
            End Set
        End Property

        Public Property EnableInheritance() As Integer
            Get
                Return m_EnableInheritance
            End Get
            Set(ByVal value As Integer)
                m_EnableInheritance = value
            End Set
        End Property
#End Region

#Region "Protected Methods"
        Protected Shadows Sub Initialize(ByVal dataRow As DataRow)
            MyBase.IdColumnName = "Information_SEQ_ID"
            MyBase.NameColumnName = "VERSION"
            MyBase.Initialize(dataRow)
            m_InformationSeqId = MyBase.GetInt(dataRow, "Information_SEQ_ID")
            m_Version = MyBase.GetString(dataRow, "VERSION")
            m_EnableInheritance = MyBase.GetInt(dataRow, "ENABLE_INHERITANCE")
        End Sub

#End Region
    End Class
End Namespace
